### ISSUE TITLE
[Request] Villager AI - Look before opening doors

### ISSUE THREAD

**Issue by ghost** [Orange1861/Improved-Villagers#23]
Ever notice that after a night of cowering in an unlit room while hordes of zombies moan and slaver outside, the first thing villagers do is open the door and walk outside? Any change of checking for nearby zombies (just a few blocks) before they open doors?

**Comment by Orange1861** [comment]
Yeah, this is an issue. I will try to fix this by making villagers go/stay inside if there are zombies nearby. This should make them more likely to survive on non-hard difficulties.

**Comment by ghost** [comment]
It's a vanilla issue, but I figured... thanks.

**Comment by Orange1861** [comment]
It is but fixing vanilla oversights (like Nitwits having no point or the unneeded 0.35 doors rules now that we have the villagers needing food) is the point of this mod. I'm not the world's best modder by far but I try to fix areas that are easyish to fix.

**Comment by ghost** [comment]
I understand. I wanted to make a mod that villager-proofed villages. Close up holes, cut down cacti (which villagers like to stand on), fill in lava, maybe prevent farms from spawning on a pillar of dirt. Alas, I didn't have the patience to unravel the Minecraft APIs.
